pkgs/development/tools/xcbuild/default.nix

```diff
@@ -39,5 +39,6 @@ in stdenv.mkDerivation rec {
     rmdir $out/usr
   '';
 
-  buildInputs = [ cmake zlib libxml2 libpng CoreServices CoreGraphics ImageIO ];
-}
\ No newline at end of file
+  buildInputs = [ cmake zlib libxml2 libpng ninja ]
+    ++ stdenv.lib.optionals stdenv.isDarwin [ CoreServices CoreGraphics ImageIO ];
```

Comment:
That's really helpful. Honestly, I'm not sure what I was expecting it to do on Linux. I suppose I thought it might be able to just compile to the host without any modification. I think we always just assume it's compiling to the native environment. Is there any way to force clang to accept macOS args and extensions but actually compile to Linux, I wonder? Not that any one on Linux is dying to use Apple's "write" command or anything but it would be really interesting to see what happens.

We do have some "cross compiling" in Nixpkgs support. I really don't understand how it works enough to set it up, though. It would be really interesting to try to link this into xcbuild.